JVM 17 & Kotlin 1.6.x Support

The changes made to the final version of 1.20.0 have reverted fixes that addressed the `invalid jvm target` error experienced in this ticket. https://github.com/detekt/detekt/issues/4287

```bash
* What went wrong:
Execution failed for task ':match-api:detekt'.
> Invalid value passed to --jvm-target
```

I'm currently unable to upgrade to resolve the current `XML External Entity (XXE) Injection` vulnerability. Version 1.20.0-RC2 continues to have no issues with JVM 17 and Kotlin 1.6.x, but does not have a resolution for the vulnerability.

The reproducer is nothing special. Its I believe the most basic possible setup: https://github.com/rocketraman/test-detekt-4786.

```
./gradlew --version

------------------------------------------------------------
Gradle 7.4.2
------------------------------------------------------------

Build time:   2022-03-31 15:25:29 UTC
Revision:     540473b8118064efcc264694cbcaa4b677f61041

Kotlin:       1.5.31
Groovy:       3.0.9
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          17.0.3 (Oracle Corporation 17.0.3+8-LTS-111)
OS:           Linux 5.16.20-200.fc35.x86_64 amd64

$ ./gradlew detekt
> Task :detekt FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':detekt'.
> Invalid value passed to --jvm-target

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 696ms
1 actionable task: 1 executed
```

Here is the build scan: https://scans.gradle.com/s/3okubnbhzpsb4

--- a/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/ArgumentConverters.kt
+++ b/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/ArgumentConverters.kt
@@ -2,7 +2,6 @@ package io.gitlab.arturbosch.detekt.cli
 
 import com.beust.jcommander.IStringConverter
 import com.beust.jcommander.ParameterException
-import org.jetbrains.kotlin.config.JvmTarget
 import org.jetbrains.kotlin.config.LanguageVersion
 import java.io.File
 import java.net.URL
@@ -51,15 +50,6 @@ class LanguageVersionConverter : IStringConverter<LanguageVersion> {
         val validValues by lazy { LanguageVersion.values().joinToString { it.versionString } }
         return requireNotNull(LanguageVersion.fromFullVersionString(value)) {
             "\"$value\" passed to --language-version, expected one of [$validValues]"
-        }
-    }
-}
-
-class JvmTargetConverter : IStringConverter<JvmTarget> {
-    override fun convert(value: String): JvmTarget {
-        val validValues by lazy { JvmTarget.values().joinToString { it.description } }
-        return requireNotNull(JvmTarget.fromString(value)) {
-            "\"$value\" passed to --jvm-target, expected one of [$validValues]"
         }
     }
 }

--- a/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/CliArgs.kt
+++ b/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/CliArgs.kt
@@ -188,11 +188,10 @@ class CliArgs {
 
     @Parameter(
         names = ["--jvm-target"],
-        converter = JvmTargetConverter::class,
         description = "EXPERIMENTAL: Target version of the generated JVM bytecode that was generated during " +
-            "compilation and is now being used for type resolution"
+            "compilation and is now being used for type resolution (1.6, 1.8, 9, 10, 11, 12, 13, 14, 15, 16 or 17)"
     )
-    var jvmTarget: JvmTarget = JvmTarget.DEFAULT
+    var jvmTarget: String = JvmTarget.DEFAULT.description
 
     @Parameter(
         names = ["--version"],

--- a/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Spec.kt
+++ b/detekt-cli/src/main/kotlin/io/gitlab/arturbosch/detekt/cli/Spec.kt
@@ -65,7 +65,7 @@ internal fun CliArgs.createSpec(output: Appendable, error: Appendable): Processi
         }
 
         compiler {
-            jvmTarget = args.jvmTarget.toString()
+            jvmTarget = args.jvmTarget
             languageVersion = args.languageVersion?.versionString
             classpath = args.classpath?.trim()
         }

